Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -1117,6 +1117,26 @@ class ConstantFoldingPhase : public Phase {
 
                 break;
             }
+
+            case HasStructureWithFlags: {
+                const AbstractValue& abstractValue = m_state.forNode(node->child1());
+                unsigned flags = node->structureFlags();
+                ASSERT(flags);
+
+                if (Structure::bitFieldFlagsCantBeChangedWithoutTransition(flags) && abstractValue.m_type && !(abstractValue.m_type & ~SpecCell) && abstractValue.m_structure.isFinite()) {
```

Comment:
This is not correct if Structure is Dictionary, correct?

Reply:
Yes, added an `isDictionary()` check to CF.